data/initial-composition/slab-model/create_slab2_database.py

```diff
@@ -0,0 +1,164 @@
+    Since the input data grid for each slab is a subset of this target grid, we can simply replace 
+    the target grid at those locations with the input slab grid. 
+    We have checked that in the input files longitude increases first and then latitude, suggesting that the 
+    elements are ordered identically to the target mesh.
+    """
+    
+    # load the input slab depth files
+    slab_data = np.loadtxt (slab_filename, delimiter=",")
+    points_x  = slab_data[:, 0] 
+    points_y  = 90 - slab_data[:, 1]
+    depths    = slab_data[:, 2]    
+    
+    x0 = np.min(points_x)
+    x1 = np.max(points_x)
+    y0 = np.min(points_y)
+    y1 = np.max(points_y)
+    
+    # need this to compare two floats
+    numerical_error = 1e-6
+   
+    indx = np.where ( (longitude_grid.flatten() >= x0 - numerical_error) & 
+                      (longitude_grid.flatten() <= x1 + numerical_error) &
+                      (latitude_grid.flatten()  >= y0 - numerical_error) & 
+                      (latitude_grid.flatten()  <= y1 + numerical_error) )
+    
+    # We do not simply replace all the points with the slab grid. This is because there 
+    # are some slabs in the western US with overlapping grid points. However, locations 
+    # in the slab grid where depths and thickness are not defined are nan values, and we
+    # want to avoid replacing existing depth and thickness values with nan values 
+    # from another slab grid.
+    # Therefore, we create a dummy variable that stores all the grid points in the slab 
+    # and then copies only the non-nan values into the target array.
+    depths_temp       = np.empty (len(longitude_grid.flatten()),)  
+    depths_temp[:]    = np.nan
+    depths_temp[indx] = slab_data[:, 2]
+    
+    # replace the target mesh with the slab data
+    indx_not_a_nan                = np.where(~np.isnan (depths_temp))
+    depths_target[indx_not_a_nan] = depths_temp[indx_not_a_nan]
+    
+    # do the same for the thickness file
+    thk_file  = slab_filename.replace ('dep', 'thk')
+    thk_data  = np.loadtxt (thk_file, delimiter=",", usecols=2)
+    
+    # We checked that slab thickness is defined at points where
+    # slab depths are defined.
+    thickness_target[indx_not_a_nan] = thk_data[np.where(~np.isnan (depths))]
+
+    return (depths_target, thickness_target)
+
+
+input_filenames = glob('*_slab2*' + file_type_key[0]  + '*.xyz')
+
+for i in range(len(input_filenames)):
+    print (input_filenames[i])
+    slab_depths, slab_thickness = slab2_into_asciiboundary (input_filenames[i])
+
+# Save the arrays in the format for aspect ascii boundary :
+# 1. use radians in longitudes and latitudes, 2. first longitudes increase then latitudes
+# 3. convert depth and thickness from km to meters 4. depth is negative in the downward
+# direction of the slab2 database, convert to positive values. 5. Add necessary header
+# information.
```

Comment:
Above it says the output is in colatitude, here it says latitude. Can you fix whichever is not correct?